tutor/src/store/courses.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice, createEntityAdapter, PayloadAction, current } from '@reduxjs/toolkit'
+import { useSelector } from 'react-redux'
+import { Course } from './types'
+import { updateCourse } from './api'
+import { bootstrap } from './bootstrap'
+
+const courseAdapter = createEntityAdapter<Course>({
+    sortComparer: (a, b) => a.name.localeCompare(b.name),
+})
+
+const initialState = courseAdapter.getInitialState()
+interface CourseSlice { courses: typeof initialState }
+
+const selectors = courseAdapter.getSelectors<CourseSlice>(s => s.courses)
+
+
+const coursesSlice = createSlice({
+    name: 'courses',
+    initialState,
+    reducers: {
+        rename(state, { payload: { id, name } }: PayloadAction<{ id: string, name: string }>) {
+            console.log(current(state))
+            const course = state.entities[id]
+            if (course) { course.name = name }
+            console.log(current(state))
+        }
+    },
+    extraReducers: (builder) => {
+        builder.addCase(updateCourse.fulfilled, (state, { payload: course }) => {
+            state.ids[course.id] = course
+        })
+        builder.addCase(bootstrap, (state, action) => {
+            courseAdapter.setAll(state, action.payload.courses)
+        })
+    }
```

Comment:
`reducers` are how the store is updated.  the plain ones only operate on the `courses` part of the store, while `extraReducers` can act on the entire store if needed